src/connection.rs

```diff
@@ -91,7 +94,17 @@ impl<IO: AsyncRead + AsyncWrite + Unpin> EppConnection<IO> {
     /// receieved to the request
     pub(crate) async fn transact(&mut self, content: &str) -> Result<String, Error> {
         debug!("{}: request: {}", self.registry, content);
-        self.send_epp_request(content).await?;
+
+        match self.send_epp_request(content).await {
+            Ok(()) => {}
+            Err(Error::Io(_)) => {
+                self.stream = self.connector.connect(self.timeout).await?;
+                self.send_epp_request(content).await?;
```

Comment:
Testing this, I think we actually need:

```suggestion
                self.stream = self.connector.connect(self.timeout).await?;
                self.greeting = self.get_epp_response().await?;
                self.send_epp_request(content).await?;
```

to pull the new greeting off the wire